File: rules/vuln_samples/positive/php/sql_injection.php
<?php
// POSITIVE: SQL injection
$id = $_GET['id'];

// Unsafe: direct query with GET param
mysqli_query($conn, "SELECT * FROM users WHERE id = '" . $id . "'");

// Unsafe: mysql_query concatenation
$db->query("SELECT * FROM products WHERE id = " . $_POST['product_id']);
?>

File: rules/vuln_samples/positive/go/sql_injection.go
package vulnsamples

import (
	"database/sql"
	"fmt"
	"net/http"
)

func getUser(w http.ResponseWriter, r *http.Request) {
	db, _ := sql.Open("sqlite3", "./app.db")
	id := r.FormValue("id")
	
	// Unsafe: fmt.Sprintf in query
	rows, _ := db.Query(fmt.Sprintf("SELECT * FROM users WHERE id = '%s'", id))
	_ = rows
	
	// Unsafe: concatenation
	db.Exec("DELETE FROM users WHERE id = '" + id + "'")
}
